### ISSUE TITLE
`fatal: --local can only be used inside a git repository` error on containers

### ISSUE THREAD

**Issue by eitsupi** [peter-evans/create-pull-request#1170]
### Subject of the issue

Maybe related to actions/checkout#760.

This action seems to fail on containers using the latest version of Git.
The checkout action had a fix for this yesterday, but I believe other actions like this one continue to be affected.

### Steps to reproduce

Use this action on containers like `ubuntu:focal`.

**Comment by peter-evans** [comment]
Hi @eitsupi 

Thank you for bringing this to my attention.

I've confirmed that the workaround provided in that issue works with this action. If you run into this issue, please use the workaround as follows.

```yml
      - uses: actions/checkout@v3

      - run: git config --global --add safe.directory "$GITHUB_WORKSPACE"

      # Make changes to pull request here

      - name: Create Pull Request
        uses: peter-evans/create-pull-request@v4
```

I will work on a fix for this issue soon.

**Comment by cat-master21** [comment]
Unfortunately the `git config --global --add safe.directory "$GITHUB_WORKSPACE"` fix does not work for too @peter-evans.

**Comment by tn3rb** [comment]
getting the same issue with the following output:

```  
Save persisted git credentials
  /usr/bin/git config --local --name-only --get-regexp http.https://github.com/.extraheader ^AUTHORIZATION:
  fatal: --local can only be used inside a git repository
Determining the base and head repositories
  /usr/bin/git config --local --get remote.origin.url
  fatal: --local can only be used inside a git repository
  Error: The format of '' is not a valid GitHub repository URL
Restore persisted git credentials
  /usr/bin/git config --local --name-only --get-regexp http.https://github.com/.extraheader ^AUTHORIZATION:
  fatal: --local can only be used inside a git repository
```

**Comment by tn3rb** [comment]
well I'm happy to report that I made some changes to my action and fixed this error.

I was using this action as part of a multi action workflow and had placed the `uses: actions/checkout@v3` step as the first step in my workflow:

```
        - uses: actions/checkout@v3

        - # other actions

        - name: Create Pull Request
          id: create-pull-request
          uses: peter-evans/create-pull-request@v4
```

This resulted in the `fatal: --local can only be used inside a git repository` error.

Moving the `uses: actions/checkout@v3` step so that it ran immediately before this action fixed the problem:

```
        - # other actions
   
     - uses: actions/checkout@v3
   
     - name: Create Pull Request
       id: create-pull-request
       uses: peter-evans/create-pull-request@v4
```

hope this maybe helps someone else out

**Comment by wmendes-ionos** [comment]
I am running into the same issue. I've tried to add `git config --global --add safe.directory "$GITHUB_WORKSPACE"` on a step earlier by the issue remains. I cannot have steps in order as suggested by @tn3rb.

Any other tip on how to overcome this ?

**Comment by peter-evans** [comment]
For those running into this error and finding this issue, as far as I'm aware, the workaround I posted is the only way to resolve it. If you cannot get your workflow to work with the workaround, then there is probably something you are doing wrong in the workflow.

The official checkout action has "fixed" this issue, but they have done it in a way where the fix doesn't persist to later workflow steps. You can see the fix here: https://github.com/actions/checkout/pull/762

> We need to figure out how to address this at an ecosystem level, outside of the checkout action.

They also acknowledge that this is an ecosystem level problem and should probably be solved there somehow. I've haven't attempted to implement a [similar fix to checkout](https://github.com/actions/checkout/pull/762) because I was hoping they would fix it at the ecosystem level.

To be clear, the "fix" that checkout have made to their action is the same as the workaround. They just do it for you inside the action and remove the config again at the end. I might implement a similar fix at some point, just to make the workaround step unnecessary, but it's not high priority.

**Comment by peter-evans** [comment]
I've created a fix for this which adds the `safe.directory` configuration for the local repository path at the start of the action, and removes it again at the end.

I'm planning to release this in the near future as part of a new major version. I would really appreciate testing/feedback for this fix. Feel free to use `v5-beta` in active workflows. I will keep that version available and make sure there is plenty of warning before it's deleted after the `v5` release.

```yml
      - uses: peter-evans/create-pull-request@v5-beta
```

**Comment by Xtrah** [comment]
We are still experiencing issues using v5-beta, getting the same error:
```
  /usr/bin/git config --local --get remote.origin.url
  fatal: --local can only be used inside a git repository
  Error: The format of '' is not a valid GitHub repository URL
```

This is our workflow:
```yml
name: Scheduled production release
on:
  schedule:
    - cron: "0 0 * * 3"

jobs:
  auto-pull-request:
    name: Create release PR to master
    runs-on: ubuntu-latest
    steps:
      - name: Create Pull Request
        uses: peter-evans/create-pull-request@v5-beta
        with:
          token: ${{ secrets.GITHUB_TOKEN }}
          title: "Weekly production release"
          body: "Weekly release"
          branch: "dev"
          base: "master"
          labels: "ignore-semantic"
```

Any tips?

**Comment by peter-evans** [comment]
@Xtrah There are a couple of things wrong with your workflow. Firstly, you don't have any checkout step. Secondly, it looks like you are trying to use an existing branch for the `branch` input. Please see the docs [here](https://github.com/peter-evans/create-pull-request/blob/main/docs/examples.md#keep-a-branch-up-to-date-with-another) for an example of what I think you are trying to do. Please make a new issue if you need further help. The problems with your workflow are unrelated to this issue. The error only matches this issue because you haven't run checkout.

**Comment by jun2zhou** [comment]
@peter-evans  try to use your workaround, there is a security violation to use your create-pull-request
[Error: .github#L1]
peter-evans/create-pull-request@v5 is not allowed to be used. Actions in this workflow must be: within a repository that belongs to your Enterprise account, created by GitHub, or verified in the GitHub Marketplace.

**Comment by peter-evans** [comment]
This is now fixed in `v5.0.0` / `v5`.

If you are using `v5-beta` or `v5-rc`, please update to `v5`. I will remove pre-release versions after 4 weeks. Thank you for testing the pre-release version!
